hedera-node/src/main/java/com/hedera/services/ledger/TransactionalLedger.java

```diff
@@ -71,18 +71,21 @@
 	private final List<K> createdKeys = new ArrayList<>(MAX_ENTITIES_LIKELY_TOUCHED_IN_LEDGER_TXN);
 	private final List<K> changedKeys = new ArrayList<>(MAX_ENTITIES_LIKELY_TOUCHED_IN_LEDGER_TXN);
 	private final List<K> perishedKeys = new ArrayList<>(MAX_ENTITIES_LIKELY_TOUCHED_IN_LEDGER_TXN);
+
 	private final Class<P> propertyType;
 	private final Supplier<A> newEntity;
 	private final BackingStore<K, A> entities;
 	private final TransactionalLedger<K, P, A> entitiesLedger;
 	private final ChangeSummaryManager<A, P> changeManager;
 	private final Function<K, EnumMap<P, Object>> changeFactory;
 
-	final Map<K, EnumMap<P, Object>> changes = new HashMap<>();
+	private final Map<K, EnumMap<P, Object>> changes = new HashMap<>();
+	private final Map<K, MerkleLeafChanges<K, A, P>> idToMerkleLeafChanges = new HashMap<>();
```

Comment:
I don't think the extra overhead of a `Map` is justified here; all we need is an `ArrayList<MerkleLeafChanges>` that we can pass to the interceptor's `preview()` method (if a non-null interceptor is set).